hivemq-edge/src/frontend/src/extensions/datahub/designer/data_policy/DataPolicyPanel.tsx

```diff
@@ -0,0 +1,45 @@
+import { FC, useMemo } from 'react'
+import { Node } from 'reactflow'
+import { Card, CardBody } from '@chakra-ui/react'
+
+import { DataPolicyData, PanelProps } from '@datahub/types.ts'
+import useDataHubDraftStore from '@datahub/hooks/useDataHubDraftStore.ts'
+import { ReactFlowSchemaForm } from '@datahub/components/forms/ReactFlowSchemaForm.tsx'
+import { MOCK_DATA_POLICY_SCHEMA } from '@datahub/designer/data_policy/DataPolicySchema.ts'
+import { CustomValidator } from '@rjsf/utils'
+import { useGetAllDataPolicies } from '@datahub/api/hooks/DataHubDataPoliciesService/useGetAllDataPolicies.tsx'
+import { useTranslation } from 'react-i18next'
+
+export const DataPolicyPanel: FC<PanelProps> = ({ selectedNode, onFormSubmit }) => {
+  const { t } = useTranslation('datahub')
+  const { nodes } = useDataHubDraftStore()
+  const { data: allPolicies } = useGetAllDataPolicies()
+
+  const data = useMemo(() => {
+    const adapterNode = nodes.find((e) => e.id === selectedNode) as Node<DataPolicyData> | undefined
+    return adapterNode ? adapterNode.data : null
+  }, [selectedNode, nodes])
+
+  const customValidate: CustomValidator<DataPolicyData> = (formData, errors) => {
+    if (!allPolicies) errors['id']?.addError(t('error.validation.dataPolicy.notLoading'))
+    else {
+      const isIdNotUnique = Boolean(allPolicies.items?.find((e) => e.id === formData?.id))
```

Comment:
2nd set of comments:
- I like how you use a `const` with a clear name so that whoever reads the code knows that it is 👍
- What this function might not catch is instances in which `formData.id` is `undefined`. Can this happen? Do you have a test about it? What do you want the code to do if formData.id is undefined and you find another undefined ID in allPolicies.items, or if formData.id is undefined and you don't find another undefined ID in allPolicies.items (hence formData.id appears to be unique)?

Reply:
formdata is the data coming from the form so `undefined` can happen indeed, when not set. 
id in allPolicies cannot be undefined (they are required both in type and in validating the form)
So if it is `undefined`, it is certainly caught by the required validation rule. If not, `undefined` cannot be used as a duplicated is so. by extension, it it unique :-)